lib/cli/writer.js

```diff
@@ -108,7 +115,12 @@ exports.writeImage = (imagePath, drive, options, onProgress) => {
           return Bluebird.resolve();
         }
 
-        return mountutils.unmountDiskAsync(drive.device);
+        // Because Mac OS doesn't like immediate unmounting / ejection
+        // after closing the file handle, as it's busy remounting,
```

Comment:
It seems to do it with every image that contains mountable partitions. I can confirm I can reproduce with `dd`